projects/sol/src/lib/sol.component.spec.ts

```diff
@@ -22,13 +22,29 @@ describe('SolComponent', () => {
     .compileComponents()
   })
 
-  beforeEach(() => {
+  beforeEach(fakeAsync(() => {
     fixture = TestBed.createComponent(SolComponent)
     component = fixture.componentInstance
+    component.mpsServer = 'wss://localhost'
+    component.authToken = 'authToken'
+    tick(4500)
     fixture.detectChanges()
-  })
+    flush()
+  }))
 
   it('should create', () => {
     expect(component).toBeTruthy()
   })
+
+  it('should clear the core objects on sol stop', () => {
+    component.stopSol()
+
+    expect(component.term).toBe(null)
+  })
+
+  it('should update the terminal state on terminal state change', () => {
+    const state = 0
+    component.onTerminalStateChange(component.redirector, state)
+    expect(component.deviceState).toEqual(state)
```

Comment:
Missing expectation that the device status was emitted.

Reply:
Updated